### PR TITLE
Restore endian_util.h and progressive directory

### PR DESCRIPTION
**Pull request by dahlstrom-g**
Issue: 380285071

### CODE REVIEW

File: media/starboard/base/endian_util.h [hunk 1]
```diff
@@ -0,0 +1,143 @@
+// Copyright 2016 The Cobalt Authors. All Rights Reserved.
```
Comment (borongc) [inline media/starboard/base/endian_util.h:1]: This is also put under `//media/starboard/progressive`.

Comment (kaidokert) [reply media/starboard/base/endian_util.h:1]: FWIF - @dahlstrom-g IMO it's okay to do "restore" commit and "move" right after, these don't need to be bundled.

Comment (borongc) [reply media/starboard/base/endian_util.h]: SGTM.

Comment (dahlstrom-g) [reply media/starboard/base/endian_util.h]: done

**Review by borongc — APPROVED** [review]

**Pull request merged**